validate_clerk_input.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Thu Aug 19 18:26:19 2021
+
+@author: vjham
+"""
+
+from product_list import products
+
+
+def validate_clerk_input(clerk_input):
+    
+    list_of_valid_product_ids = [str(product['id']) for product in products]
```

Comment:
This line of code is going to execute every time the clerk inputs any text. That seems like overkill. This list of product ids seems like something we only need to generate once, since the list of valid ids is very unlikely to change while the clerk is inputting items.

Let me know if you need help with this!

Also - excellent use of list comprehension!

Reply:
Thank you, and that is an excellent point! I think I do need some help with this.

My thought is that it would make the most sense if the list of valid product ids was in product_list.py, but that would not rectify the problem you have illustrated here.

Reply:
I have been doing some python research and I think we will want to use this @cached_property decorator https://docs.python.org/3/library/functools.html?highlight=cached_property#functools.cached_property